src/classes/worker.ts

```diff
@@ -77,10 +77,12 @@ export class Worker<T = any> extends QueueBase {
     }
     this.timerManager = new TimerManager();
 
-    /* tslint:disable: no-floating-promises */
-    this.run().catch(error => {
-      console.error(error);
-    });
+    if (!this.opts.settings || !this.opts.settings.disableAutoRun) {
```

Comment:
Does this still work? :D

Wouldn't you want to make sure `this.opts.settings` exists first before checking `disableAutoRun`?

Reply:
Makes me wonder if its also possible to run a quick unit test for this case :)

Reply:
Yup. `!this.opts.settings` basically returns true if `this.ops.settings` does not exist. So then the `or` condition early evaluates to true. If `this.opts.settings` does exist, then `!this.opts.settings` will evaluate to false and we'll check the next condition. The way I switched it though is using `A && B == !A || !B`. Added a test to be sure.